test/builders/build/substrate-api/polkadotjs-api.js

```diff
@@ -0,0 +1,176 @@
+  const ethDerPath = "m/44'/60'/0'/0/" + index;
+
+  const getApi = async () => {
+    // Construct API provider
+    const wsProvider = new WsProvider(process.env.WSS_RPC_ENDPOINT);
+    const api = await ApiPromise.create({ provider: wsProvider });
+
+    return api;
+  };
+
+  describe("Querying for Information - State Queries", async () => {
+    it("should return the account balance and nonce", async () => {
+      const api = await getApi();
+
+      const bob = ethers.Wallet.createRandom().address;
+      const { nonce, data: balance } = await api.query.system.account(bob);
+
+      assert.equal(nonce, 0n);
+      assert.equal(balance.free, 0n);
+    });
+  });
+
+  describe("Querying for Information - RPC Queries", async () => {
+    it("should return the chain name", async () => {
+      const api = await getApi();
+
+      const chain = await api.rpc.system.chain();
+
+      assert.equal(chain, "Moonbase Development Testnet");
+    });
+    it("should return the last block number and hash", async () => {
+      const api = await getApi();
+
+      const lastHeader = await api.rpc.chain.getHeader();
+
+      // Cannot predict the hash or block number but we can test that a value is returned
+      assert.exists(lastHeader.number);
+      assert.exists(lastHeader.hash);
+    });
+  });
+
+  describe("Keyrings - Adding Accounts", async () => {
+    it("should extract the ethereum address from the mnemonic", async () => {
+      // Create a keyring instance
+      const keyringECDSA = new Keyring({ type: "ethereum" });
+
+      // Extract Ethereum address from mnemonic
+      const newPairEth = keyringECDSA.addFromUri(
+        `${alice.mnemonic}/${ethDerPath}`
+      );
+
+      assert.equal(newPairEth.address, alice.address);
+    });
+    it("should extract the private key from the mnemonic", async () => {
+      // Extract private key from mnemonic
+      const privateKey = u8aToHex(
+        hdEthereum(
+          mnemonicToLegacySeed(alice.mnemonic, "", false, 64),
+          ethDerPath
+        ).secretKey
+      );
+
+      assert.equal(privateKey, alice.pk);
+    });
+    it("should extract the address from the private key", async () => {
+      // Create a keyring instance
+      const keyringECDSA = new Keyring({ type: "ethereum" });
+
+      // Extract address from private key
+      const otherPair = keyringECDSA.addFromUri(alice.pk);
+
+      assert.equal(otherPair.address, alice.address);
+    });
+  });
+
+  describe("Transactions - Sending Basic Transactions", async () => {
+    it("should sign and send a transaction successfully", async () => {
+      const api = await getApi();
+
+      // Create a keyring instance
+      const keyring = new Keyring({ type: "ethereum" });
+
+      // Initialize wallet key pairs
+      const aliceFromUri = keyring.addFromUri(alice.pk);
+      const bob = ethers.Wallet.createRandom().address;
+
+      // Form the transaction
+      const tx = api.tx.balances.transfer(bob, 12345);
```

Comment:
it is always good practice to use big numbers for everything, not critical here tho